scripts/deployment/phase2-assets/collaterals/deploy_yearn_v2_curve_usdc.ts

```diff
@@ -0,0 +1,103 @@
+import fs from 'fs'
+import hre from 'hardhat'
+import { getChainId } from '../../../../common/blockchain-utils'
+import { networkConfig } from '../../../../common/configuration'
+import { bn, fp } from '../../../../common/numbers'
+import { expect } from 'chai'
+import { CollateralStatus } from '../../../../common/constants'
+import {
+  getDeploymentFile,
+  getAssetCollDeploymentFilename,
+  IAssetCollDeployments,
+  getDeploymentFilename,
+  fileExists,
+} from '../../common'
+import { priceTimeout, oracleTimeout } from '../../utils'
+import { YearnV2CurveFiatCollateral } from '../../../../typechain'
+import { ContractFactory } from 'ethers'
+import { YVUSDC_LP_TOKEN } from '../../../../test/plugins/individual-collateral/yearnv2/constants'
+
+async function main() {
+  // ==== Read Configuration ====
+  const [deployer] = await hre.ethers.getSigners()
+
+  const chainId = await getChainId(hre)
+
+  console.log(`Deploying Collateral to network ${hre.network.name} (${chainId})
+    with burner account: ${deployer.address}`)
+
+  if (!networkConfig[chainId]) {
+    throw new Error(`Missing network configuration for ${hre.network.name}`)
+  }
+
+  // Get phase1 deployment
+  const phase1File = getDeploymentFilename(chainId)
+  if (!fileExists(phase1File)) {
+    throw new Error(`${phase1File} doesn't exist yet. Run phase 1`)
+  }
+  // Check previous step completed
+  const assetCollDeploymentFilename = getAssetCollDeploymentFilename(chainId)
+  const assetCollDeployments = <IAssetCollDeployments>getDeploymentFile(assetCollDeploymentFilename)
+
+  const deployedCollateral: string[] = []
+
+  /********  Deploy Yearn V2 Curve Fiat Collateral - yvCurveUSDCcrvUSD  **************************/
+
+  const YearnV2CurveCollateralFactory: ContractFactory = await hre.ethers.getContractFactory(
+    'YearnV2CurveFiatCollateral'
+  )
+
+  const collateral = <YearnV2CurveFiatCollateral>await YearnV2CurveCollateralFactory.connect(
+    deployer
+  ).deploy(
+    {
+      priceTimeout: priceTimeout.toString(),
+      chainlinkFeed: networkConfig[chainId].chainlinkFeeds.USDC, // not used but can't be empty
+      oracleError: fp('0.0025').toString(), // not used but can't be empty
+      erc20: networkConfig[chainId].tokens.yvCurveUSDCcrvUSD,
+      maxTradeVolume: fp('1e6').toString(), // $1m,
+      oracleTimeout: oracleTimeout(chainId, '86400').toString(), // 24hr -- max of all oracleTimeouts
+      targetName: hre.ethers.utils.formatBytes32String('USD'),
+      defaultThreshold: fp('0.015').toString(), // 1.5% = max oracleError + 1%
+      delayUntilDefault: bn('86400').toString(), // 24h
+    },
+    fp('1e-3').toString(), // revenueHiding = 0.1%, high for Yearn to tolerate small strategy losses
```

Comment:
we should run the backtester on this to confirm this much hiding is needed. i have a feeling it could be a couple orders of magnitude lower

Reply:
This isn't how I would think about it -- the question is not "what is the largest drawdown that has occurred historically" it is "what is the largest drawdown that might reasonably happen in the future that should not result in the collateral being defaulted".

Reply:
AFAIK, Yearn stable vaults are up-only especially the curve vaults since the initial LP is never lost/touched. However, I don't quite remember on how I know this, so maybe it's worth running the backtester anyway to see if there's a drawdown.

Reply:
I searched through the discord when doing my research for this and they indicated it is exceedingly unlikely the rate would fall but it is not up-only. An individual strategy might be up-only, but not all strategies are this way.